Cubical/Codata/M-types/stream.agda

```diff
@@ -0,0 +1,30 @@
+{-# OPTIONS --cubical --guardedness --safe #-}
+
+module Cubical.Codata.M-types.stream where
+
+open import Cubical.Data.Unit
+
+open import Cubical.Foundations.Prelude
+
+open import Cubical.Codata.M-types.M-type
+open import Cubical.Codata.M-types.helper
+open import Cubical.Codata.M-types.Container
+
+--------------------------------------
+-- Stream definitions using M-types --
+--------------------------------------
+
+stream-S : ∀ A -> Container
+stream-S A = (A , (λ _ → Unit))
+
+stream : ∀ (A : Set₀) -> Set₀
+stream A = M-type (stream-S A)
+
+cons : ∀ {A} -> A -> stream A -> stream A
+cons x xs = in-fun (x , λ { tt -> xs } )
+
+hd : ∀ {A} -> stream A -> A
+hd {A} S = out-fun S .fst
+
+tl : ∀ {A} -> stream A -> stream A
+tl {A} S = out-fun S .snd tt
```

Comment:
How about proving the equivalence of this definition of streams with https://github.com/agda/cubical/blob/master/Cubical/Codata/Stream/Base.agda ?

Reply:
Yes, I almost have a proof of this in my local development, but it is not complete yet, there are some issues with termination. Should I add it to the pull request? or make a seperate pull request for it?

Reply:
Please make another PR! I really prefer many small self-contained PRs as it makes the reviewing a lot easier